From d23b000c6b2b16a5d83f24785b4e3e44257a719b Mon Sep 17 00:00:00 2001
From: Eilo <eilo2518@gmail.com>
Date: Fri, 16 Dec 2011 00:19:25 -0500
Subject: [PATCH 05/65] 2.2.2 ICC: Sindragosa no tumba en pet

---
 .../Northrend/IcecrownCitadel/boss_sindragosa.cpp  |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp b/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp
index fb04d9c..3bca7ce 100644
--- a/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp
+++ b/src/server/scripts/Northrend/IcecrownCitadel/boss_sindragosa.cpp
@@ -1182,7 +1182,7 @@ class FrostBombTargetSelector
 
         bool operator()(Unit* unit)
         {
-            if (unit->HasAura(SPELL_ICE_TOMB_DAMAGE))
+            if (unit->HasAura(SPELL_ICE_TOMB_DAMAGE) || unit->GetTypeId() != TYPEID_PLAYER)
                 return true;
 
             for (std::list<Creature*>::const_iterator itr = _collisionList.begin(); itr != _collisionList.end(); ++itr)
-- 
1.7.8.msysgit.0

