From 4a0574347b702b638ca9c5a024fa1691f5c2335c Mon Sep 17 00:00:00 2001
From: Eilo <eilo2518@gmail.com>
Date: Fri, 16 Dec 2011 00:22:48 -0500
Subject: [PATCH 06/65] 2.2.3 ICC: Valithria adds salen hacia ella

---
 .../IcecrownCitadel/boss_valithria_dreamwalker.cpp |   34 ++++++++++++++++++--
 1 files changed, 31 insertions(+), 3 deletions(-)

diff --git a/src/server/scripts/Northrend/IcecrownCitadel/boss_valithria_dreamwalker.cpp b/src/server/scripts/Northrend/IcecrownCitadel/boss_valithria_dreamwalker.cpp
index b1b514f..69dc8cc 100644
--- a/src/server/scripts/Northrend/IcecrownCitadel/boss_valithria_dreamwalker.cpp
+++ b/src/server/scripts/Northrend/IcecrownCitadel/boss_valithria_dreamwalker.cpp
@@ -691,6 +691,12 @@ class npc_risen_archmage : public CreatureScript
                 _canCallEnterCombat = true;
             }
 
+            void IsSummonedBy(Unit* /*summoner*/)
+            {
+                if (Creature* valithria = ObjectAccessor::GetCreature(*me, _instance->GetData64(DATA_VALITHRIA_DREAMWALKER)))
+                    me->GetMotionMaster()->MoveChase(valithria);
+            }
+
             void EnterCombat(Unit* /*target*/)
             {
                 me->FinishSpell(CURRENT_CHANNELED_SPELL, false);
@@ -789,7 +795,7 @@ class npc_blazing_skeleton : public CreatureScript
 
         struct npc_blazing_skeletonAI : public ScriptedAI
         {
-            npc_blazing_skeletonAI(Creature* creature) : ScriptedAI(creature)
+            npc_blazing_skeletonAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript())
             {
             }
 
@@ -800,6 +806,12 @@ class npc_blazing_skeleton : public CreatureScript
                 _events.ScheduleEvent(EVENT_LEY_WASTE, urand(15000, 20000));
             }
 
+            void IsSummonedBy(Unit* /*summoner*/)
+            {
+                if (Creature* valithria = ObjectAccessor::GetCreature(*me, _instance->GetData64(DATA_VALITHRIA_DREAMWALKER)))
+                    AttackStart(valithria);
+            }
+
             void UpdateAI(uint32 const diff)
             {
                 if (!UpdateVictim())
@@ -833,6 +845,7 @@ class npc_blazing_skeleton : public CreatureScript
 
         private:
             EventMap _events;
+            InstanceScript* const _instance;
         };
 
         CreatureAI* GetAI(Creature* creature) const
@@ -914,7 +927,7 @@ class npc_blistering_zombie : public CreatureScript
 
         struct npc_blistering_zombieAI : public ScriptedAI
         {
-            npc_blistering_zombieAI(Creature* creature) : ScriptedAI(creature)
+            npc_blistering_zombieAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript())
             {
             }
 
@@ -923,6 +936,12 @@ class npc_blistering_zombie : public CreatureScript
                 DoCast(me, SPELL_ACID_BURST, true);
             }
 
+            void IsSummonedBy(Unit* /*summoner*/)
+            {
+                if (Creature* valithria = ObjectAccessor::GetCreature(*me, _instance->GetData64(DATA_VALITHRIA_DREAMWALKER)))
+                    AttackStart(valithria);
+            }
+
             void UpdateAI(uint32 const /*diff*/)
             {
                 if (!UpdateVictim())
@@ -930,6 +949,8 @@ class npc_blistering_zombie : public CreatureScript
 
                 DoMeleeAttackIfReady();
             }
+        private:
+            InstanceScript* const _instance;
         };
 
         CreatureAI* GetAI(Creature* creature) const
@@ -945,7 +966,7 @@ class npc_gluttonous_abomination : public CreatureScript
 
         struct npc_gluttonous_abominationAI : public ScriptedAI
         {
-            npc_gluttonous_abominationAI(Creature* creature) : ScriptedAI(creature)
+            npc_gluttonous_abominationAI(Creature* creature) : ScriptedAI(creature), _instance(creature->GetInstanceScript())
             {
             }
 
@@ -955,6 +976,12 @@ class npc_gluttonous_abomination : public CreatureScript
                 _events.ScheduleEvent(EVENT_GUT_SPRAY, urand(10000, 13000));
             }
 
+            void IsSummonedBy(Unit* /*summoner*/)
+            {
+                if (Creature* valithria = ObjectAccessor::GetCreature(*me, _instance->GetData64(DATA_VALITHRIA_DREAMWALKER)))
+                    AttackStart(valithria);
+            }
+
             void JustDied(Unit* /*killer*/)
             {
                 DoCast(me, SPELL_ROT_WORM_SPAWNER, true);
@@ -988,6 +1015,7 @@ class npc_gluttonous_abomination : public CreatureScript
 
         private:
             EventMap _events;
+            InstanceScript* const _instance;
         };
 
         CreatureAI* GetAI(Creature* creature) const
-- 
1.7.8.msysgit.0

